src/containers/App.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-param-reassign */
+import React, { PureComponent } from 'react';
+import styled from 'styled-components';
+import { moviesData, filtersData } from '../data';
+import Header from '../components/Header';
+import MoviesResult from './MoviesResult';
+import Footer from '../components/Footer';
+import ErrorBoundary from './ErrorBounding';
+
+class App extends PureComponent {
+    // eslint-disable-next-line react/state-in-constructor
+    state = {
+        searchValue: '',
+        sortValue: 'date',
+        filters: filtersData,
+        all: true,
+        movies: moviesData
+    }
+
+    searchOnChange = (event) => {
+        this.setState({
+            searchValue: event.target.value
+        });
+    }
+
+    dynamicSearch = () => {
+        const { movies, searchValue } = this.state;
+
+        const searchMovies = searchValue
+            ? movies.filter(movie => movie.title.toLowerCase().includes(searchValue.toLowerCase()))
+            : moviesData;
+
+        return searchMovies;
+    }
+
+    handleSearch = () => {
+        const searchResult = this.dynamicSearch();
+        this.setState({ movies: searchResult });
+    };
+
+    handleKeyDown = (event) => {
+        if (event.key === 'Enter') {
+            this.handleSearch();
+        }
+    }
+
+    sortingOnChange = (event) => {
+        this.setState({
+            sortValue: event.target.value,
+        });
+    };
+
+    setAll = () => {
+        const { filters } = this.state;
+
+        filters.forEach(filter => {
+            filter.status = false;
+        });

```

Comment:
in this case you mutate data. Better to use functionals methods and return new variables.
```jsx
setAll = () => {
    this.setState({
        all: true,
        filters: this.state.filters.map(filter => ({ ...filter, status: false }))
    })
}
```